viam-cartographer.go

```diff
@@ -97,46 +93,108 @@ func init() {
 		}, &slamConfig.AttrConfig{})
 }
 
-// runtimeServiceValidation ensures the service's data processing and saving is valid for the subAlgo and
-// cameras given.
-func runtimeServiceValidation(
+// New returns a new slam service for the given robot.
+func New(
 	ctx context.Context,
-	cams []camera.Camera,
-	cartoSvc *cartographerService,
-) error {
-	if !cartoSvc.useLiveData {
-		return nil
+	deps registry.Dependencies,
+	config config.Service,
+	logger golog.Logger,
+	bufferSLAMProcessLogs bool,
+	executableName string,
+	sensorValidationMaxTimeoutSec int,
```

Comment:
[opt] I preferred using functions to set these values, since I think this parameter list is getting too long.

Reply:
How do you feel about another struct being used as options? Maybe something like a `TimeoutConfig`? I agree that this list is getting too long, however the old solution had some non-pretty global state.

Reply:
This became too complicated, provided I extracted much of the code and that it is used in multiple places. I was not confident that the old approach would not lead to unexpected behavior. I agree that the parameter list is too long, as well. Would creating a struct to collect those variables into one variable be a valid alternative?